### PR TITLE
 📖 Improve clarity of prowjob configuration regeneration instructions

### PR DESCRIPTION
**Pull request by Sunnatillo**
**What this PR does / why we need it**:
Adds more explanation on how to generate prowjob configuration on release-tasks.md file

Area example:
/area documentation

### CODE REVIEW

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -489,7 +489,7 @@ While we add test coverage for the new release branch we will also drop the test
             * Change intervals (let's use the same as for `release-1.5`).
 2. Create a new dashboard for the new branch in: `test-infra/config/testgrids/kubernetes/sig-cluster-lifecycle/config.yaml` (`dashboard_groups` and `dashboards`).
 3. Remove old release branches and unused versions from the `cluster-api-prowjob-gen.yaml` file in [test-infra](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api/) according to our policy documented in [Support and guarantees](../../CONTRIBUTING.md#support-and-guarantees). For example, let's assume we just added `release-1.6`, then we can now drop test coverage for the `release-1.3` branch.
-4. Regenerate the probjob configuration using `make generate-test-infra-prowjobs`.
+4. Regenerate the prowjob configuration running `make generate-test-infra-prowjobs` command from cluster-api repository. Before running this command, ensure to export the `TEST_INFRA_DIR` variable, specifying the location of the [test-infra](https://github.com/kubernetes/test-infra/) repository in your environment. For further information, refer to this [link](https://github.com/kubernetes-sigs/cluster-api/pull/9937).
```
[the remark this answers is not in the document]
Comment (Sunnatillo) [reply docs/release/release-tasks.md:492]: Added

**Review by adilGhaffarDev — APPROVED** [review]

**Comment by sbueringer** [comment]
Totally fine as is, but in general let's try to not link too much to PRs for documentation and rather properly document stuff. It's a bit annoying to have to go through PR diffs to gather additional context.

**Pull request merged**